plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2453,9 +2458,19 @@ public MaterializedViewFreshness getMaterializedViewFreshness(ConnectorSession s
             // Information missing
             return new MaterializedViewFreshness(false);
         }
-        Map<String, String> tableToSnapshotIdMap = Splitter.on(',').withKeyValueSeparator('=').split(dependsOnTables);
-        for (Map.Entry<String, String> entry : tableToSnapshotIdMap.entrySet()) {
-            List<String> strings = Splitter.on(".").splitToList(entry.getKey());
+        Iterable<String> tableToSnapshotIds = Splitter.on(',').split(dependsOnTables);
+        for (String entry : tableToSnapshotIds) {
+            if (entry.equals(UNKNOWN_SNAPSHOT_TOKEN)) {
+                // This is a "federated" materialized view (spanning across connectors). Trust user's choice and assume "fresh or fresh enough".
```

Comment:
Once a storage table is created from non-iceberg sources, does this mean that subsequent REFRESHes will always be a no-op since the MV is always "fresh" ? If so, is there some way to force refresh for such an MV ?

Reply:
Good catch. I've added test coverage for that. Not solved yet.